tests/vite-app/run-tests.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash -e
+
+run_test() {
+    START_SERVER_CMD=$1
+    SERVER_PORT=$2
+    START_CYPRESS_CMD=$3
+
+    echo "Starting Vite server in background ..."
+    $START_SERVER_CMD &
+    VITE_PID=$!
+    echo "Vite server launched as PID ${VITE_PID}"
+    while ! nc -z localhost $SERVER_PORT; do
```

Comment:
How do we handle a stuck server boot process , for ex in a port conflict scenario?

Reply:
Yes it would run forever until the build times out (or if run by developer until they CTRL-C).
Acceptable since this is just test runner code (?)

Reply:
I would still prefer if we put in an upper time bound , for ex `while ! nc -z localhost $SERVER_PORT; [[ "$SECONDS" -lt 2000 ]]; do` , assuming that the server boot up should not take more thant 2000 seconds since the shell has been spawned

Reply:
Done (capped at 30 seconds, which should be ample)